epoxy-adapter/src/main/java/com/airbnb/epoxy/OnModelVisibilityStateChangedListener.java

```diff
@@ -0,0 +1,58 @@
+package com.airbnb.epoxy;
+
+import android.support.annotation.IntDef;
+
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+
+/** Used to register an onVisibilityChanged callback with a generated model. */
+public interface OnModelVisibilityStateChangedListener<T extends EpoxyModel<?>, V> {
+
+  @Retention(RetentionPolicy.SOURCE)
+  @IntDef({VISIBLE, INVISIBLE, FOCUSED_VISIBLE, UNFOCUSED_VISIBLE, FULL_IMPRESSION_VISIBLE})
+  @interface VisibilityState {
```

Comment:
nesting the VisibilityState interface inside OnModelVisibilityStateChangedListener makes it quite verbose to reference state's like `VISIBLE`. what do you think of moving the `VisibilityState` interface to be top level?

I want to be careful that the naming isn't too generic to have it mixed up with other names people might use in their apps - maybe `EpoxyVisibility` or `ModelVisibility` could be used